php_phongo.c

```diff
@@ -1546,6 +1546,19 @@ static bool php_phongo_apply_options_to_uri(mongoc_uri_t* uri, bson_t* options T
 			continue;
 		}
 
+		/* Check replica set and throw an exception on an empty value */
+		if (!strcasecmp(key, MONGOC_URI_REPLICASET)) {
+			if (!BSON_ITER_HOLDS_UTF8(&iter)) {
+				PHONGO_URI_INVALID_TYPE(iter, "string");
+				return false;
+			}
+
+			if (!strcmp("", bson_iter_utf8(&iter, NULL))) {
+				phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Value for URI option \"%s\" cannot be empty string.", key);
+				return false;
+			}
+		}
```

Comment:
I think this would be better placed within the `if (mongoc_uri_option_is_utf8(key)) {` condition further down in `php_phongo_apply_options_to_uri`. That already does the `BSON_ITER_HOLDS_UTF8` check, which you're duplicating here. Consider:

```c
if (mongoc_uri_option_is_utf8(key)) {
    if (!BSON_ITER_HOLDS_UTF8(&iter)) {
        PHONGO_URI_INVALID_TYPE(iter, "string");
        return false;
    }

    if (!strcasecmp(key, MONGOC_URI_REPLICASET) && !strcmp("", bson_iter_utf8(&iter, NULL))) {
        phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Value for URI option \"%s\" cannot be empty string.", key);
        return false;
    }

    if (!mongoc_uri_set_option_as_utf8(uri, key, bson_iter_utf8(&iter, NULL))) {
        /* Assignment uses mongoc_uri_set_appname() for the "appname"...
```